pennylane/devices/null_qubit.py

```diff
@@ -285,7 +285,9 @@ def execute(self, circuit, **kwargs):
         self.apply(circuit.operations, rotations=self._get_diagonalizing_gates(circuit), **kwargs)
 
         if self.tracker.active:
-            self.tracker.update(executions=1, shots=self._shots)
+            self.tracker.update(
+                executions=1, shots=self._shots, resources=circuit.specs["resources"]
+            )
```

Comment:
Do we have to do this on a device-by-device basis, there is no way we could have the tracker track resources for all devices?

Reply:
The tracker itself was implemented like this to begin with (each device has its own update call). All I did was add the `resources` kwarg. I think it should be possible (probably 3-5 pts of work) to re-implement the tracker so that it gets called somewhere in `execute.py` before we dispatch to each device.

I am just hesitant because I don't know the execution pipeline for every device, I worry that there maybe some processing of tapes that happens between the qnode execute call and the device execute call. To me, it makes sense that the tracker is as close to the device execution function as possible, since ultimately you want to track resources for the tapes just before they get executed on the device.

Reply:
Thanks @Jaybsoni! Something we can add to the backlog.